src/Eshopworld.Core/BaseEvent.cs

```diff
@@ -21,27 +22,46 @@ internal BaseEvent()
         {
         }
 
-        /// <summary>
-        /// Converts this POCO to a <see cref="IDictionary{TKey,TValue}"/> by using JSonConvert twice (both directions).
-        /// </summary>
-        /// <returns>The converted <see cref="IDictionary{String, String}"/>.</returns>
-        [NotNull]
-        internal virtual IDictionary<string, string> ToStringDictionary()
+
+        internal  IDictionary<string, string> ToUnionStringDictionary(object adjunctObject)
+        {
+            return ToStringDictionaryInner(this).Union(ToStringDictionaryInner(adjunctObject))
+                .ToDictionary(k => k.Key, v => v.Value);
+        }
+
+        private IDictionary<string, string> ToStringDictionaryInner(object target = null)
         {
+            target = target ?? this;
+
             try
             {
-                return JsonConvert.DeserializeObject<Dictionary<string, string>>(JsonConvert.SerializeObject(this));
+                return FullFlow();
             }
             catch (Exception)
             {
-                return JsonConvert.DeserializeObject<Dictionary<string, string>>(
-                    JsonConvert.SerializeObject(this, new JsonSerializerSettings
+                return ShallowFlow();
+            }
+
+            IDictionary<string, string> FullFlow() =>
```

Comment:
Why local methods that are used once? this is a lot harder to read then just inlining the contents of the local methods.